Wikipedia/Code/NSUserDefaults+WMFExtensions.swift

```diff
@@ -116,10 +116,17 @@ let WMFSendUsageReports = "WMFSendUsageReports"
     
     @objc var themeAnalyticsName: String {
         let name = string(forKey: WMFAppThemeName)
+        let darkMode =  bool(forKey: "SystemDarkMode")
         guard name != nil, name != Theme.defaultThemeName else {
+            let theme = Theme.defaultAnalyticsThemeName
+            return darkMode ? Theme.dark.name : theme
+        }
+        
+        if Theme.withName(name)?.name == Theme.light.name {
             return Theme.defaultAnalyticsThemeName
         }
-        return Theme.withName(name)?.name ?? Theme.light.name
+        
+        return Theme.withName(name)?.name ?? Theme.defaultAnalyticsThemeName
```

Comment:
This is working great! I did find one odd instance of the name "dark-dimmed" and "black-dimmed" sending if you have the "dim images" toggle checked in the Reading preferences settings and you change the theme from an article view or the editor theming popovers. This is likely already an existing bug but we should fix it while we're here.

While I was digging around I also noticed an old `analyticsName` property on Theme.swift that appears to be unused. Because of this "-dimmed" bug though, I think it has value in us using it again. If you [update](https://github.com/wikimedia/wikipedia-ios/blob/7fd42e1d1e178d9f732574101dcf6598f937058d/WMF%20Framework/Theme.swift#L839) the light theme's `analyticsName` to be Theme.defaultAnalyticsThemeName, I think this method could then become something like:

```
guard name != nil, name != Theme.defaultThemeName else {
    return systemDarkMode ? Theme.black.analyticsName : Theme.light.analyticsName
}

return Theme.withName(name)?.analyticsName ?? Theme.light.analyticsName
```

This way this way the analytics value can hopefully be a bit more stable than the theme name, which apparently changes with that dimming setting.